qadence/mitigations/readout.py

```diff
@@ -25,51 +57,76 @@ def renormalize_counts(corrected_counts: npt.NDArray, n_shots: int) -> npt.NDArr
         corrected_counts -= counts_diff
         corrected_counts = np.where(corrected_counts < 0, 0, corrected_counts)
         sum_corrected_counts = sum(corrected_counts)
-        if sum_corrected_counts < n_shots:
-            renormalization_factor = max(sum_corrected_counts, n_shots) / min(
-                sum_corrected_counts, n_shots
-            )
-        else:
-            renormalization_factor = min(sum_corrected_counts, n_shots) / max(
-                sum_corrected_counts, n_shots
-            )
+
+        renormalization_factor = n_shots / sum_corrected_counts
         corrected_counts = np.rint(corrected_counts * renormalization_factor).astype(int)
     return corrected_counts
 
 
 def mitigation_minimization(
-    noise: Noise, mitigation: Mitigations, samples: list[Counter]
+    noise: Noise,
+    mitigation: Mitigations,
+    samples: list[Counter],
 ) -> list[Counter]:
     """Minimize a correction matrix subjected to stochasticity constraints.
 
     See Equation (5) in https://arxiv.org/pdf/2001.09980.pdf.
+    See Page(3) in https://arxiv.org/pdf/1106.5458.pdf for MLE implementation
+
+    Args:
+        noise: Specifies confusion matrix and default error probability
+        mitigation: Selects additional mitigation options based on noise choice.
+        For readout we have the following mitigation options for optimization
+        1.constrained_opt, 2.mle. Default : mle
+        samples: List of samples to be mitigated
+
+    Returns:
+        Mitigated counts computed by the algorithm
     """
     noise_matrices = noise.options.get("noise_matrix", noise.options["confusion_matrices"])
+    optimization_type = mitigation.options.get("optimization_type", "mle")
```

Comment:
Right, instead of retrieving the optimization type, I would pass it as an argument to the function with a default value. That would mean deciding which of the two minimizations we want as default. Then retrieving it in the mitigate function below.

Reply:
I havent set a default. I am not fully convinced one is better than the other always in terms of divergence or the 2_norm probability difference achieved.